nifi-nar-bundles/nifi-iceberg-bundle/nifi-iceberg-processors/src/test/java/org/apache/nifi/processors/iceberg/TestPutIcebergCustomValidation.java

```diff
@@ -0,0 +1,129 @@
+        runner.enableControllerService(readerFactory);
+
+        runner.setProperty(PutIceberg.RECORD_READER, RECORD_READER_NAME);
+    }
+
+    private void initCatalogService(Configuration configuration) throws InitializationException {
+        TestHiveCatalogService catalogService = new TestHiveCatalogService.Builder().withConfig(configuration).build();
+
+        runner.addControllerService(CATALOG_SERVICE_NAME, catalogService);
+        runner.enableControllerService(catalogService);
+
+        runner.setProperty(PutIceberg.CATALOG, CATALOG_SERVICE_NAME);
+    }
+
+    private void initKerberosUserService() throws InitializationException {
+        KerberosUserService kerberosUserService = mock(KerberosUserService.class);
+        when(kerberosUserService.getIdentifier()).thenReturn(KERBEROS_USER_SERVICE_NAME);
+
+        runner.addControllerService(KERBEROS_USER_SERVICE_NAME, kerberosUserService);
+        runner.enableControllerService(kerberosUserService);
+
+        runner.setProperty(PutIceberg.KERBEROS_USER_SERVICE, KERBEROS_USER_SERVICE_NAME);
+    }
+
+    @Test
+    public void testCustomValidateWithKerberosSecurityConfigAndWithoutKerberosUserService() throws InitializationException {
+        initRecordReader();
+
+        Configuration config = new Configuration();
+        config.set("hadoop.security.authentication", "kerberos");
+        initCatalogService(config);
+
+        runner.setProperty(PutIceberg.CATALOG_NAMESPACE, CATALOG_NAMESPACE);
+        runner.setProperty(PutIceberg.TABLE_NAME, TABLE_NAME);
+        runner.assertNotValid();
+    }
+
+    @Test
+    public void testCustomValidateWithKerberosSecurityConfigAndKerberosUserService() throws InitializationException {
+        initRecordReader();
+
+        Configuration config = new Configuration();
+        config.set("hadoop.security.authentication", "kerberos");
+        initCatalogService(config);
+
+        initKerberosUserService();
+
+        runner.setProperty(PutIceberg.CATALOG_NAMESPACE, CATALOG_NAMESPACE);
+        runner.setProperty(PutIceberg.TABLE_NAME, TABLE_NAME);
+        runner.assertValid();
+    }
+
+    @Test
+    public void testCustomValidateWithoutKerberosSecurityConfigAndKerberosUserService() throws InitializationException {
+        initRecordReader();
+
+        initCatalogService(new Configuration());
+
+        runner.setProperty(PutIceberg.CATALOG_NAMESPACE, CATALOG_NAMESPACE);
+        runner.setProperty(PutIceberg.TABLE_NAME, TABLE_NAME);
+        runner.assertValid();
+    }
+
+    @Test
+    public void testCustomValidateWithoutKerberosSecurityConfigAndWithKerberosUserService() throws InitializationException {
+        initRecordReader();
+
+        initCatalogService(new Configuration());
+
+        initKerberosUserService();
+
+        runner.setProperty(PutIceberg.CATALOG_NAMESPACE, CATALOG_NAMESPACE);
+        runner.setProperty(PutIceberg.TABLE_NAME, TABLE_NAME);
+        runner.assertValid();
+    }
```

Comment:
This config (`hadoop.security.authentication` is not kerberos but having `KerberosUserService` configured on the processor) does not work so I think the processor should be invalid in this case.

Even worse, the processor cannot recover after unsetting `KerberosUserService`. I think it is because the kerberos state is stored and stuck in hadoop's static context. Other hadoop related processors handle it via adding `@RequiresInstanceClassLoading`.
I also tested it with a non-kerberized and a kerberized PutIceberg processor in the same flow (connecting to different Hive clusters, unsecured and secured, respectively) and the two processors interfere. After running the kerberized one, the non-kerberized cannot work.
Could you please create a separate bug ticket for it and address the issue?

Reply:
Thanks @turcsanyip, nice catch on the processors interference. I've created a ticket to fix that issue: [NIFI-11334](https://issues.apache.org/jira/browse/NIFI-11334)

I've changed the validation for the scenario where the KerberosUserService is provided but the security authentication is not set to 'kerberos'.